### PR TITLE
Add jenkins file

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -79,15 +80,15 @@ pipeline {
...
+                sh 'mvn -U -B -e clean install -DskipTests -T1C'
             }
         }
 
         stage('Tests') {
             steps {
                 echo 'Running tests'
                 // all tests is very very long (10 hours on Apache Jenkins)
-                sh 'mvn -B -e test'
+                sh 'mvn -B -e test -T1C'
```
Comment (chibenwa) [inline Jenkinsfile:91]: That's probably ambitious for test phase.

Comment (mbaechler) [reply Jenkinsfile:91]: Agree. As isolation is poor we may try to leverage Jenkins plugins to split tests between different runners to keep good isolation level.
Let's first look at the result of this try (:

Comment (chibenwa) [reply Jenkinsfile:91]: Also we might blow up the test runner as parallel runs are achieved at the maven module level.

Of course, I'd be happy to not rely on surefire forks and rely on maven concurrency instead.

Thanks for your work on that with @jeantil BTW.

**Comment by chibenwa** [comment]
Fingers crossed !

**Comment by jeantil** [comment]
I wonder if this is another flaky test, at least the run went a bit further this time 
build>5 stopped at  Apache James :: Mailbox :: Cassandra ............... FAILURE [09:02 min]
build>6 stopped at  Apache James :: Mailbox :: Event :: RabbitMQ implementation FAILURE [03:28 min]
```
[INFO] Apache James :: Mailbox :: Cassandra ............... SUCCESS [08:09 min]
[INFO] Apache James :: Mailbox :: Tika .................... SUCCESS [ 23.728 s]
[INFO] Apache James :: Mailbox :: ElasticSearch ........... SUCCESS [03:50 min]
[INFO] Apache James :: Mailbox :: Event :: JSON ........... SUCCESS [  7.754 s]
[INFO] Apache James :: Mailbox :: Event :: In Cassandra implementation SUCCESS [ 59.285 s]
[INFO] Apache James :: Mailbox :: Event :: RabbitMQ implementation FAILURE [03:28 min
```
so there is progress !

Now to test the issueCommentTrigger :

test this please

**Comment by mbaechler** [comment]
test this please

**Comment by jeantil** [comment]
```
issueCommentTrigger
> Currently this trigger will only allow collaborators of the repository in question to trigger builds.
```
I wonder who is a collaborator of the repository then since the trigger didn't work for either of us.

**Comment by mbaechler** [comment]
replaced by https://github.com/apache/james-project/pull/268

**Pull request closed without merging**